torch/utils/data/datapipes/iter/dataframes.py

```diff
@@ -0,0 +1,416 @@
+
+    def __iter__(self):
+        buffer = []
+        for df in self.source_datapipe:
+            buffer.append(df)
+            if len(buffer) == self.batch:
+                yield pandas.concat(buffer)
+                buffer = []
+        if len(buffer):
+            yield pandas.concat(buffer)
+
+
+@functional_datapipe('dataframes_shuffle', is_df=True)
+class ShuffleDataFramesPipe(DFIterDataPipe):
+    def __init__(self, source_datapipe):
+        self.source_datapipe = source_datapipe
+        if not WITH_PANDAS:
+            Exception('DataFrames prototype requires pandas to function')
+
+    def __iter__(self):
+        size = None
+        all_buffer = []
+        for df in self.source_datapipe:
+            if size is None:
+                size = len(df.index)
+            for i in range(len(df.index)):
+                all_buffer.append(df[i:i + 1])
+        random.shuffle(all_buffer)
+        buffer = []
+        for df in all_buffer:
+            buffer.append(df)
+            if len(buffer) == size:
+                yield pandas.concat(buffer)
+                buffer = []
+        if len(buffer):
+            yield pandas.concat(buffer)
+
+
+@functional_datapipe('dataframes_filter', is_df=True)
+class ShuffleDataFramesPipe(DFIterDataPipe):
+    def __init__(self, source_datapipe, filter_fn):
+        self.source_datapipe = source_datapipe
+        self.filter_fn = filter_fn
+        if not WITH_PANDAS:
+            Exception('DataFrames prototype requires pandas to function')
+
+    def __iter__(self):
+        size = None
+        all_buffer = []
+        filter_res = []
+        for df in self.source_datapipe:
+            if size is None:
+                size = len(df.index)
+            for i in range(len(df.index)):
+                all_buffer.append(df[i:i + 1])
+                filter_res.append(self.filter_fn(df.iloc[i]))
+
+        buffer = []
+        for df, res in zip(all_buffer, filter_res):
+            if res:
+                buffer.append(df)
+                if len(buffer) == size:
+                    yield pandas.concat(buffer)
+                    buffer = []
+        if len(buffer):
+            yield pandas.concat(buffer)
+
+
+@functional_datapipe('to_dataframes_pipe', is_df=True)
+class ExampleAggregateAsDataFrames(DFIterDataPipe):
+    def __init__(self, source_datapipe, dataframe_size=10, columns=None):
+        self.source_datapipe = source_datapipe
+        self.columns = columns
+        self.dataframe_size = dataframe_size
+        if not WITH_PANDAS:
+            Exception('DataFrames prototype requires pandas to function')
+
+    def _as_list(self, item):
+        try:
+            return [i for i in item]
+        except:
+            return [item]
+
+    def __iter__(self):
+        aggregate = []
+        for item in self.source_datapipe:
+            aggregate.append(self._as_list(item))
+            if len(aggregate) == self.dataframe_size:
```

Comment:
`len(aggregate)` can be larger than `dataframe_size` here. We need to yield first `dataframe_size` of items, and keep the rest in the buffer